haystack/nodes/prompt/invocation_layer/amazon_bedrock.py

```diff
@@ -0,0 +1,134 @@
+import json
+import logging
+from abc import abstractmethod, ABC
+from typing import Optional, Dict, Union, List, Any
+
+
+from haystack.errors import AmazonBedrockConfigurationError
+from haystack.lazy_imports import LazyImport
+from haystack.nodes.prompt.invocation_layer import PromptModelInvocationLayer
+from haystack.nodes.prompt.invocation_layer.handlers import DefaultPromptHandler
+
+logger = logging.getLogger(__name__)
+
+
+with LazyImport(message="Run 'pip install farm-haystack[aws]'") as boto3_import:
+    import boto3
+    from botocore.exceptions import ClientError, BotoCoreError
+
+
+class AmazonBedrockBaseInvocationLayer(PromptModelInvocationLayer, ABC):
+    """
+    Base class for Amazon Bedrock based invocation layers.
+    """
+
+    def __init__(
+        self,
+        model_name_or_path: str,
+        aws_access_key_id: Optional[str] = None,
+        aws_secret_access_key: Optional[str] = None,
+        aws_session_token: Optional[str] = None,
+        aws_region_name: Optional[str] = None,
+        aws_profile_name: Optional[str] = None,
+        max_length: Optional[str] = 2048,
+        **kwargs,
+    ):
+        super().__init__(model_name_or_path, **kwargs)
+        self.max_length = max_length
+
+        try:
+            session = AmazonBedrockBaseInvocationLayer.create_session(
+                aws_access_key_id=aws_access_key_id,
+                aws_secret_access_key=aws_secret_access_key,
+                aws_session_token=aws_session_token,
+                aws_region_name=aws_region_name,
+                aws_profile_name=aws_profile_name,
+            )
+            self.client = session.client("bedrock-runtime")
+        except:
+            raise AmazonBedrockConfigurationError
+
+    def _ensure_token_limit(self, prompt: Union[str, List[Dict[str, str]]]) -> Union[str, List[Dict[str, str]]]:
+        # the prompt for this model will be of the type str
+        print(
+            "Tokenizer for the bedrock models are not available publicly. The tokens will get truncated automatically"
+        )
+        return prompt
+
+    @classmethod
+    def supports(cls, model_name_or_path, **kwargs):
+        if model_name_or_path in [
+            "amazon.titan-text-express-v1",
+            "amazon.titan-text-lite-v1",
+            "ai21.j2-ultra-v1",
+            "ai21.j2-mid-v1",
+            "cohere.command-text-v14",
+        ]:
```

Comment:
We should also support `anthropic.claude-v2`, `anthropic.claude-instant-v1` and `anthropic.claude-v1`.

Reply:
Additionally we should be able to use the `boto3.list_foundation_models()` method (see https://docs.aws.amazon.com/bedrock/latest/userguide/api-methods-list.html) ?
The endpoint can be queried by OutputModality `TEXT` (see https://docs.aws.amazon.com/bedrock/latest/APIReference/API_ListFoundationModels.html)

Reply:
Sure @tstadel, adding that in the next commit.